cmd/sops/common/common.go

```diff
@@ -251,6 +252,22 @@ func LoadEncryptedFileWithBugFixes(opts GenericDecryptOpts) (*sops.Tree, error)
 		}
 	}
 
+	awsProfile := os.Getenv("AWS_PROFILE")
+	if opts.UseAwsProfile != "" {
+		awsProfile = opts.UseAwsProfile
```

Comment:
I don't think this is the right place to look at environment variables. This is a library function deep down the call tree.

Maybe adding `EnvVar: "AWS_PROFILE",` to the CLI option would be a better place to handle this? This might have other unintended consequences, though.

Reply:
Agreed. I can move this logic up in the stack.